### DISCUSSION TITLE
Access to the repos

### DISCUSSION THREAD

**Discussion by rnadupal** [community/community#111036] [Enterprise] [answered]
A Noob question. Looking for recommendations on how to add new users to GHE to an existing team group automatically?

**Comment by lmdevv** [comment, marked as answer]
Its pretty straightforward with some Python and Github Actions. A general approach would be something like:
1. **Use the GitHub API:** It's a powerful way to automate a lot of stuff, including users in team groups.
2. **Automation Script:** Write a small script (most probably Python) that calls the GitHub API to add users to your specified team. This can be triggered manually or set up to run at regular intervals.
3. **GitHub Actions:** For a more integrated solution, consider setting up a GitHub Action that listens for specific events (like new user creation) and triggers your user-adding script. You can also use Python cronjobs to run the script regularly

Python script would look something like:
```python
import requests

# Your GitHub Enterprise host
GITHUB_API = ''
TOKEN = ''
ORG_NAME = ''
TEAM_SLUG = ''
USERNAME = ''

headers = {
    'Authorization': f'token {TOKEN}',
    'Accept': 'application/vnd.github.v3+json',
}

def add_user_to_team(username):
    url = f'{GITHUB_API}/orgs/{ORG_NAME}/teams/{TEAM_SLUG}/memberships/{username}'
    response = requests.put(url, headers=headers)
    if response.status_code == 200:
        print(f'Successfully added {username} to the team.')
    else:
        print(f'Failed to add {username} to the team. Status code: {response.status_code}')

add_user_to_team(USERNAME)

```
